src/main/java/org/springframework/data/elasticsearch/client/elc/CriteriaQueryProcessor.java

```diff
@@ -397,4 +432,48 @@ public static String escape(String s) {
 		return sb.toString();
 	}
 
+    /**
+     * Convert a spring-data-elasticsearch {@literal query} to an Elasticsearch {@literal query}.
+     * <p>
+     * The reason it was copied from {@link RequestConverter#getQuery(org.springframework.data.elasticsearch.core.query.Query, Class)} is because of a circular dependency.
```

Comment:
I don't think we should do that duplication. If both `RequestConverter` and `CriteriaQueryProcessor` need that code, it should be extracted into a class that is used by both.